content/blog/2022/2022-06-03-distribute-static-content-with-cloudfront.md

```diff
@@ -0,0 +1,257 @@
+
+{{% image alt="browser" src="images/posts/aws-cloudfront/browser-cf.png" %}}
+
+## Securing Access to Content
+In the earlier sections, we used static assets residing in a public S3 bucket which makes it insecure by making all the content accessible to users if the S3 bucket URL is known to them. CloudFront provides many configurations to secure access to content. For this example, we will use an Origin Access Identity (OAI) to restrict access to the contents of the S3 bucket. 
+
+Origin Access Identity (OAI) is a special CloudFront user that is associated with our distributions. We can restrict access to the S3 bucket by updating the bucket policy to provide read permission to the OAI defined as the `Principal` in the policy definition as shown below:
+
+```json
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+            "Sid": "2",
+            "Effect": "Allow",
+            "Principal": {
+                "AWS": "arn:aws:iam::cloudfront:user/CloudFront Origin Access Identity <OAI>"
+            },
+            "Action": "s3:GetObject",
+            "Resource": "arn:aws:s3:::<S3 bucket name>/*"
+        }
+  ]
+}
+```
+Let us create another CloudFront Distribution but configured to use an OAI to access the contents in the S3 bucket:
+{{% image alt="browser" src="images/posts/aws-cloudfront/oai.png" %}}
+
+This time we have chosen the S3 bucket URL from the selection box as the origin domain instead of the bucket website endpoint. In the section for S3 bucket access, we have selected `Yes use OAI` and created an OAI: `my-oai` to associate with this distribution. We have also chosen the option of updating the bucket policy manually after creating the distribution.
+
+After creating the distribution, let us update the bucket policy of our S3 bucket to look like this:
+
+```json
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+            "Sid": "1",
+            "Effect": "Allow",
+            "Principal": {
+                "AWS": "arn:aws:iam::cloudfront:user/CloudFront Origin Access Identity E32V87I09SD18I"
+            },
+            "Action": "s3:GetObject",
+            "Resource": "arn:aws:s3:::io.myapp/*"
+    }
+  ]
+}
+```
+This bucket policy grants the CloudFront origin access identity (OAI) with id: `E32V87I09SD18I` permission to get (read) all objects in our Amazon S3 bucket. We have set the `Principal` to the OAI id which can be found from the [AWS management console](https://console.aws.amazon.com/cloudfront/v3/home#/oai). 
+
+We have also disabled the public access to the bucket and the static web hosting property. 
+
+After the CloudFront distribution is deployed and active, we can navigate to our website using the CloudFront distribution domain name: `https://d4l1ajcygy8jp.cloudfront.net/index.html`:
+
+{{% image alt="browser" src="images/posts/aws-cloudfront/browser-cf-oai.png" %}}
```

Comment:
How the SSL connection is configured? Automatically by CloudFront? Please explain.